src/Core/Manifest/VersionProvider.php

```diff
@@ -40,15 +49,85 @@ class VersionProvider
     public function getVersion()
     {
         $modules = $this->getModules();
+        $this->updateModulesWithLockFileRecipes($modules);
         $lockModules = $this->getModuleVersionFromComposer(array_keys($modules));
-        $output = [];
+
+        $arr = [];
+        $unknownVersion = _t(__CLASS__ . '.VERSIONUNKNOWN', 'Unknown');
         foreach ($modules as $module => $title) {
-            $version = isset($lockModules[$module])
-                ? $lockModules[$module]
-                : _t(__CLASS__ . '.VERSIONUNKNOWN', 'Unknown');
-            $output[] = $title . ': ' . $version;
+            $version = $lockModules[$module] ?? $unknownVersion;
+            $arr[$module] = "$title: $version";
+        }
+        $this->filterPriorityModules($arr);
+        return implode(', ', array_values($arr));
+    }
+
+    /**
+     * Update modules to include two silverstripe recipes, if they exist in the composer.lock file
+     *
+     * Hard-coding here instead of adding to respective module _config/config.yml
+     * because recipes are not loaded, instead contents of modules app/_config/*.yml
+     * are copied to the project app/_config folder, and it would be annoying for usres to have
+     * 2x recipe-config.yml files copied into their projects _config folder just for version labels
+     *
+     * @param $modules
+     * @return void
+     */
+    private function updateModulesWithLockFileRecipes(&$modules): void
+    {
+        $data = [
+            'silverstripe/recipe-core' => 'Core Recipe',
+            'silverstripe/recipe-cms' => 'CMS Recipe',
+        ];
+        $lockData = $this->getComposerLock();
+        foreach ($data as $module => $label) {
+            // user has custom config for recipe label
+            if (array_key_exists($module, $modules)) {
+                continue;
+            }
+            // error getting composer.lock file
+            if (!$lockData || empty($lockData['packages'])) {
+                continue;
+            }
+            // validate hardcoded recipe exists in composer.lock file before adding to modules
+            foreach ($lockData['packages'] as $package) {
+                if ($module != $package['name']) {
+                    continue;
+                }
+                $modules[$module] = $label;
+                break;
+            }
+        }
+    }
+
+    /**
+     * Filter array of modules to only show a priority module if it exists
+     *
+     * Priority modules are hardcoded for similar reasons as updateModulesWithLockModuleRecipes()
+     *
+     * @param array $arr
+     * @return void
+     */
+    private function filterPriorityModules(array &$arr): void
+    {
+        $priorityModules = [
+            'cwp/cwp-core',
```

Comment:
Rather than putting CWP references into the core cor you can use extension point.

Reply:
I don't believe so? I don't think this would work for recipe-cms / recipe-core since they're not loaded as modules i.e. vendor/silverstripe/recipe-cms does not have a _config folder or config.php

That's kind of why this PR is done with all the hard-coding instead of the usual thing where modules are basically not aware of each other